error[E0277]: `[(); _]` is not a future
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ---------------------------^^^^^^
   |                 |                          |
   |                 |                          `[(); _]` is not a future
   |                 |                          help: remove the `.await`
   |                 this call returns `[(); _]`
   |
   = help: the trait `Future` is not implemented for `[(); _]`
   = note: [(); _] must be a future or must implement `IntoFuture` to be awaited
   = note: required for `[(); _]` to implement `IntoFuture`

error[E0698]: type inside `async` block must be known in this context
  --> $DIR/unresolved-ct-var.rs:6:17
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ^^^^^^^^^^^^^^^^^^^ cannot infer the value of const parameter `N` declared on the function `from_fn`
   |
note: the type is part of the `async` block because of this `await`
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                                            ^^^^^^

error[E0698]: type inside `async` block must be known in this context
  --> $DIR/unresolved-ct-var.rs:6:17
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ^^^^^^^^^^^^^^^^^^^ cannot infer the value of const parameter `N` declared on the function `from_fn`
   |
note: the type is part of the `async` block because of this `await`
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                                            ^^^^^^

error[E0698]: type inside `async` block must be known in this context
  --> $DIR/unresolved-ct-var.rs:6:17
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ^^^^^^^^^^^^^^^^^^^ cannot infer the value of const parameter `N` declared on the function `from_fn`
   |
note: the type is part of the `async` block because of this `await`
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                                            ^^^^^^

error[E0698]: type inside `async` block must be known in this context
  --> $DIR/unresolved-ct-var.rs:6:17
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ^^^^^^^^^^^^^^^^^^^ cannot infer the value of const parameter `N` declared on the function `from_fn`
   |
note: the type is part of the `async` block because of this `await`
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                                            ^^^^^^

error[E0698]: type inside `async` block must be known in this context
  --> $DIR/unresolved-ct-var.rs:6:17
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                 ^^^^^^^^^^^^^^^^^^^ cannot infer the value of const parameter `N` declared on the function `from_fn`
   |
note: the type is part of the `async` block because of this `await`
  --> $DIR/unresolved-ct-var.rs:6:44
   |
LL |         let s = std::array::from_fn(|_| ()).await;
   |                                            ^^^^^^

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0698.
For more information about an error, try `crablangc --explain E0277`.
